### PR TITLE
Migrate Internal Pipeline list API from rails to spark

### PR DESCRIPTION
**Pull request by viraj2712**
Issue- #6661

### CODE REVIEW

**Comment by ketan** [comment]
Does the pipeline structure api work instead of this?

**Comment by viraj2712** [comment]
Looks like `Pipeline Structure API` can be used by non-admin users as well, whereas `Pipelines API` is only for admin and group admin users.

Also, there is change in response of both the APIs:
- Response of `Pipeline Structure API` returns list of pipelines grouped by pipeline groups and templates.
- Response of `Pipelines API` returns list of pipelines.

File: api/api-internal-pipelines-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelines/InternalPipelinesControllerV1.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("", mimeType, this::setContentType);
+            before("/*", mimeType, this::setContentType);
+
+            before("", mimeType, this.apiAuthenticationHelper::checkAdminUserOrGroupAdminUserAnd403);
+
+            get("", mimeType, this::index);
+
+            exception(HttpException.class, (Exception exception, Request request, Response response) -> {
```
Comment (bdpiparva) [inline api/api-internal-pipelines-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelines/InternalPipelinesControllerV1.java]: No need to do this. See `HttpException.class` for more information.

File: api/api-internal-pipelines-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelines/representers/PipelineConfigsWithMinimalAttributesRepresenter.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import com.thoughtworks.go.config.PipelineConfig;
+import com.thoughtworks.go.config.PipelineConfigs;
+import com.thoughtworks.go.spark.Routes;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class PipelineConfigsWithMinimalAttributesRepresenter {
+    public static void toJSON(OutputWriter outputWriter, List<PipelineConfigs> listOfPipelineConfigs) {
+        List<PipelineConfig> pipelines = listOfPipelineConfigs.stream().flatMap(pipelineConfigs -> pipelineConfigs.getPipelines().stream()).collect(Collectors.toList());
```
Comment (bdpiparva) [inline api/api-internal-pipelines-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelines/representers/PipelineConfigsWithMinimalAttributesRepresenter.java]: Do we really need stream in `pipelineConfigs -> pipelineConfigs.getPipelines().stream()`?

Comment (viraj2712) [reply api/api-internal-pipelines-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelines/representers/PipelineConfigsWithMinimalAttributesRepresenter.java]: done

Comment (viraj2712) [reply api/api-internal-pipelines-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelines/representers/PipelineConfigsWithMinimalAttributesRepresenter.java]: `flatMap` requires `mapper element` as a `Stream` which we pass as second argument
Refactored

File: api/api-internal-pipelines-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelines/InternalPipelinesControllerV1.java (hunk 1 above)
Comment (viraj2712) [reply api/api-internal-pipelines-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelines/InternalPipelinesControllerV1.java]: Removed the exception handling code

**Pull request merged**